src/frame/worker/crypto/recrypt/tests/RecryptWasm.test.ts

```diff
@@ -52,6 +52,48 @@ describe("RecryptWasm", () => {
         });
     });
 
+    describe("rotateGroupPrivateKeyWithRetry", () => {
+        const groupPrivateKey = new Uint8Array([22, 33, 44]);
+        it("should result in an error when mocked generateKeyPair returns Uint8Array of zeros for new privateKey", () => {
+            jest.spyOn(Recrypt.getApi(), "generatePlaintext");
+            Recrypt.rotateGroupPrivateKeyWithRetry(groupPrivateKey).engage(
+                (error) => {
+                    expect(error.message).toEqual("Key rotation failed.");
+                    expect(Recrypt.getApi().generatePlaintext).toHaveBeenCalledTimes(2);
+                },
+                () => fail("Should not success when operation fails")
+            );
+        });
+
+        it("should result in an error when subtracting existing private key from the new private key", () => {
+            jest.spyOn(Recrypt.getApi(), "generatePlaintext").mockReturnValue(new Uint8Array([12, 23, 34]) as any);
+            jest.spyOn(MockRecrypt, "subtractPrivateKeys").mockReturnValue(new Uint8Array(32));
+            Recrypt.rotateGroupPrivateKeyWithRetry(groupPrivateKey).engage(
+                (error) => {
+                    expect(error.message).toEqual("Key rotation failed.");
```

Comment:
Similar to above, it'd be better to have some assertion here about the calls to `generatePlaintext` or `subtractPrivateKeys` instead of asserting this exact error message text.

Reply:
done